### PR TITLE
Add iwyu for Linux

### PR DESCRIPTION
**Pull request by mattdurak**

### CODE REVIEW

File: build/devops_gated.yml [hunk 1]
```diff
@@ -266,3 +266,28 @@ jobs:
...
+  pool:
+    name: Azure-MessagingStore
+    demands:
+      - linux
+
+  steps:
+
+  - bash: |
+     pushd $(Build.Repository.LocalPath)
+     git submodule update --init
```
Comment (dcristoloveanu) [inline build/devops_gated.yml:280]: >git submodule update --init [](start = 5, length = 27)

Interesting, does this work for Linux? We had creds issues for Windows with it.
Maybe the machine has the credentials saved somehow :-)

Comment (dcristoloveanu) [reply build/devops_gated.yml:280]: For Windows I added submodules:true

---
In reply to: [564146457](https://github.com/Azure/c-pal/pull/69#discussion_r564146457) [](ancestors = 564146457)

File: common/tests/call_once_int/call_once_int.c [hunk 2]
```diff
@@ -2,17 +2,22 @@
 // Licensed under the MIT license. See LICENSE file in the project root for full license information.
 
 #ifdef __cplusplus
+#include <cstdint>
 #include <cstdlib>
 #include <ctime>
 #else
+#include <stdbool.h>
+#include <stdint.h>
```
Comment (dcristoloveanu) [inline common/tests/call_once_int/call_once_int.c:10]: I am confused. Isn't this also building with iwyu on Win?

File: common/tests/call_once_ut/call_once_ut.c [hunk 3]
```diff
@@ -2,18 +2,18 @@
...
-#include <cstdlib>
 #else
-#include <stdlib.h>
+#include <stdbool.h>
 #endif
 
-#include "macro_utils/macro_utils.h"
+#include "macro_utils/macro_utils.h" // IWYU pragma: keep
 
+// IWYU pragma: no_include <wchar.h>
```
Comment (dcristoloveanu) [inline common/tests/call_once_ut/call_once_ut.c:11]: >no_include [](start = 16, length = 10)

Oooo, I learned something new

File: build/devops_gated.yml (hunk 1 above)
Comment (mattdurak) [reply build/devops_gated.yml:280]: copy-pasta from the normal Linux job above, so it works I guess?

---
In reply to: [564146540](https://github.com/Azure/c-pal/pull/69#discussion_r564146540) [](ancestors = 564146540,564146457)

File: common/tests/call_once_int/call_once_int.c (hunk 2 above)
Comment (mattdurak) [reply common/tests/call_once_int/call_once_int.c:10]: Jelani asked the same about my umock changes. Not sure why iwyu misses some of these with Clang+cl.exe but catches them for Linux Clang 6.0...

---
In reply to: [564146942](https://github.com/Azure/c-pal/pull/69#discussion_r564146942) [](ancestors = 564146942)

**Review by dcristoloveanu — APPROVED** [review]
:shipit:

**Pull request merged**